File: Backend-Java/src/main/java/com/example/backend/dto/RegularisationRequestDTO.java
package com.example.backend.dto;

import java.util.Date;

public class RegularisationRequestDTO {
    private Date targetDate;
    private String targetTime; // e.g., "09:30" - for missed entry
    private String type; // "Clock-in" or "Clock-out" - for missed entry
    private String newClockInTime; // e.g., "09:30" - for modifying existing log
    private String newClockOutTime; // e.g., "17:30" - for modifying existing log
    private String reason;
    private String attendanceLogId; // For modifying existing log

    // Getters and Setters

    public Date getTargetDate() {
        return targetDate;
    }

    public void setTargetDate(Date targetDate) {
        this.targetDate = targetDate;
    }

    public String getTargetTime() {
        return targetTime;
    }

    public void setTargetTime(String targetTime) {
        this.targetTime = targetTime;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getNewClockInTime() {
        return newClockInTime;
    }

    public void setNewClockInTime(String newClockInTime) {
        this.newClockInTime = newClockInTime;
    }

    public String getNewClockOutTime() {
        return newClockOutTime;
    }

    public void setNewClockOutTime(String newClockOutTime) {
        this.newClockOutTime = newClockOutTime;
    }

    public String getReason() {
        return reason;
    }

    public void setReason(String reason) {
        this.reason = reason;
    }

    public String getAttendanceLogId() {
        return attendanceLogId;
    }

    public void setAttendanceLogId(String attendanceLogId) {
        this.attendanceLogId = attendanceLogId;
    }
}

File: FrontEnd/flutter_application/lib/models/regularisation_request_dto.dart
class RegularisationRequestDTO {
  final DateTime targetDate;
  final String? targetTime; // e.g., "09:30" - for missed entry
  final String? type; // "Clock-in" or "Clock-out" - for missed entry
  final String? newClockInTime;
  final String? newClockOutTime;
  final String reason;
  final String? attendanceLogId;

  RegularisationRequestDTO({
    required this.targetDate,
    this.targetTime,
    this.type,
    this.newClockInTime,
    this.newClockOutTime,
    required this.reason,
    this.attendanceLogId,
  });

  Map<String, dynamic> toJson() {
    return {
      'targetDate': targetDate.toIso8601String(),
      'targetTime': targetTime,
      'type': type,
      'newClockInTime': newClockInTime,
      'newClockOutTime': newClockOutTime,
      'reason': reason,
      'attendanceLogId': attendanceLogId,
    };
  }
}
